src/cloudai/schema/test_template/nemo_run/slurm_command_gen_strategy.py

```diff
@@ -67,10 +69,18 @@ def generate_test_command(
 
         command = ["nemo", "llm", cmd_args_dict.pop("task"), "--factory", cmd_args_dict.pop("recipe_name"), "-y"]
 
-        if tr.nodes:
-            command.append(f"trainer.num_nodes={len(self.system.parse_nodes(tr.nodes))}")
-        elif tr.num_nodes > 0:
-            command.append(f"trainer.num_nodes={tr.num_nodes}")
+        num_nodes = len(self.system.parse_nodes(tr.nodes)) if tr.nodes else tr.num_nodes
+
+        try:
+            if cmd_args_dict["trainer"]["num_nodes"] != num_nodes:
```

Comment:
Why can't we override it this value?

P.S. `tdef.cmd_args.trainer.num_nodes != num_nodes` looks better and allows type checking.